lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceManagerStatus.java

```diff
@@ -147,7 +149,7 @@ private synchronized void onRMDone(final RuntimeStatusEvent runtimeStatusEvent)
     this.driverStatusManager.onComplete();
   }
 
-  private synchronized void onRMRunning(final RuntimeStatusEvent runtimeStatusEvent) {
+  private void onRMRunning(final RuntimeStatusEvent runtimeStatusEvent) {
```

Comment:
It is true onRMRunning and isIdle() have the same lock, but onPotentiallyIdle() is locked on DriverIdleManager. Remove the lock on onRMRunning  will break the lock on isIdle() and onPotentiallyIdle() as a bundle so that whoever comes first will get the lock in sequence rather than deadlock.